src/client/client4.ts

```diff
@@ -807,210 +876,210 @@ export default class Client4 {
-        return this.doFetch(
+    getStatusesByIds = (userIds: string[]) => {
+        return this.doFetch<UserStatus[]>(
             `${this.getUsersRoute()}/status/ids`,
             {method: 'post', body: JSON.stringify(userIds)},
         );
     };
 
-    getStatus = async (userId: string) => {
-        return this.doFetch(
+    getStatus = (userId: string) => {
+        return this.doFetch<UserStatus>(
             `${this.getUserRoute(userId)}/status`,
             {method: 'get'},
         );
     };
 
-    updateStatus = async (status: UserStatus) => {
-        return this.doFetch(
+    updateStatus = (status: UserStatus) => {
+        return this.doFetch<UserStatus>(
             `${this.getUserRoute(status.user_id)}/status`,
             {method: 'put', body: JSON.stringify(status)},
         );
     };
 
-    switchEmailToOAuth = async (service: string, email: string, password: string, mfaCode = '') => {
+    switchEmailToOAuth = (service: string, email: string, password: string, mfaCode = '') => {
         this.trackEvent('api', 'api_users_email_to_oauth');
 
-        return this.doFetch(
+        return this.doFetch<{follow_link: string}>(
             `${this.getUsersRoute()}/login/switch`,
             {method: 'post', body: JSON.stringify({current_service: 'email', new_service: service, email, password, mfa_code: mfaCode})},
         );
     };
 
-    switchOAuthToEmail = async (currentService: string, email: string, password: string) => {
+    switchOAuthToEmail = (currentService: string, email: string, password: string) => {
         this.trackEvent('api', 'api_users_oauth_to_email');
 
-        return this.doFetch(
+        return this.doFetch<{follow_link: string}>(
             `${this.getUsersRoute()}/login/switch`,
             {method: 'post', body: JSON.stringify({current_service: currentService, new_service: 'email', email, new_password: password})},
         );
     };
 
-    switchEmailToLdap = async (email: string, emailPassword: string, ldapId: string, ldapPassword: string, mfaCode = '') => {
+    switchEmailToLdap = (email: string, emailPassword: string, ldapId: string, ldapPassword: string, mfaCode = '') => {
         this.trackEvent('api', 'api_users_email_to_ldap');
 
-        return this.doFetch(
+        return this.doFetch<{follow_link: string}>(
             `${this.getUsersRoute()}/login/switch`,
             {method: 'post', body: JSON.stringify({current_service: 'email', new_service: 'ldap', email, password: emailPassword, ldap_id: ldapId, new_password: ldapPassword, mfa_code: mfaCode})},
         );
     };
 
-    switchLdapToEmail = async (ldapPassword: string, email: string, emailPassword: string, mfaCode = '') => {
+    switchLdapToEmail = (ldapPassword: string, email: string, emailPassword: string, mfaCode = '') => {
         this.trackEvent('api', 'api_users_ldap_to_email');
 
-        return this.doFetch(
+        return this.doFetch<{follow_link: string}>(
```

Comment:
Since we're redefining this type over a few times, should we just create a type for it?